src/main/java/github/scarsz/discordsrv/DiscordSRV.java

```diff
@@ -528,6 +529,15 @@ public void connectFailed(URI arg0, SocketAddress arg1, IOException arg2) {}
             token = token.replaceAll("[^\\w\\d-_.]", "");
         }
 
+        final ThreadFactory callbackThreadFactory = new ThreadFactoryBuilder().setNameFormat("DiscordSRV - JDA Callback").build();
+        final ExecutorService callbackThreadPool = Executors.newFixedThreadPool(Integer.min(Runtime.getRuntime().availableProcessors(), 10), callbackThreadFactory);
```

Comment:
`ForkJoinPool#commonPool` is technically a size of `availableProcessors - 1` and maximum of `0x7fff` (`32767`) OR decided by evironment variable `java.util.concurrent.ForkJoinPool.common.parallelism`. Probably should have atleast a bit higher of a maximum

A `ForkJoinPool` could be used instead of `ThreadPoolExecutor`. See benafits of [the ForkJoinPool](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ForkJoinPool.html) and [ForkJoinTasks](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ForkJoinTask.html)